### ISSUE TITLE
Handle mercure error exception

### ISSUE THREAD

**Issue by MarcHagen** [shlinkio/shlink#1385]
#### How Shlink is set-up

* Shlink Version: 3.0.2
* PHP Version: Docker Image
* How do you serve Shlink: Docker image
* Database engine used: MariaDB (10.6.x)

#### Summary

It seems that a mercure exception is not handled correctly? And is floading the logs with big text logs.

#### Current behavior

```shell
4g4lk [2022-02-19T21:01:18.785403+00:00] [e7c0d6ef-d9d0-4c45-9410-c4b71fbbdfc3] Access.INFO - - "GET /rest/v2/health HTTP/1.1" 200 121
4g4lk [2022-02-19T21:01:18.849838+00:00] [3d892419-b884-4d60-a829-6d103be3444a] Shlink.ERROR - Shlinkio\Shlink\Rest\Exception\MercureException: This Shlink instance is not integrated with a mercure hub. in /etc/shlink/module/Rest/src/Exception/MercureException.php:21
4g4lk Stack trace:
4g4lk #0 /etc/shlink/module/Rest/src/Action/MercureInfoAction.php(30): Shlinkio\Shlink\Rest\Exception\MercureException::mercureNotConfigured()
4g4lk #1 /etc/shlink/vendor/laminas/laminas-stratigility/src/Middleware/RequestHandlerMiddleware.php(46): Shlinkio\Shlink\Rest\Action\MercureInfoAction->handle(Object(Laminas\Diactoros\ServerRequest))
4g4lk #2 /etc/shlink/vendor/mezzio/mezzio/src/Middleware/LazyLoadingMiddleware.php(37): Laminas\Stratigility\Middleware\RequestHandlerMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #3 /etc/shlink/vendor/laminas/laminas-stratigility/src/Next.php(51): Mezzio\Middleware\LazyLoadingMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #4 /etc/shlink/vendor/laminas/laminas-stratigility/src/MiddlewarePipe.php(76): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
4g4lk #5 /etc/shlink/vendor/mezzio/mezzio-router/src/Route.php(85): Laminas\Stratigility\MiddlewarePipe->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #6 /etc/shlink/vendor/mezzio/mezzio-router/src/RouteResult.php(97): Mezzio\Router\Route->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #7 /etc/shlink/vendor/mezzio/mezzio-router/src/Middleware/DispatchMiddleware.php(30): Mezzio\Router\RouteResult->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #8 /etc/shlink/vendor/mezzio/mezzio/src/Middleware/LazyLoadingMiddleware.php(37): Mezzio\Router\Middleware\DispatchMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #9 /etc/shlink/vendor/laminas/laminas-stratigility/src/Next.php(51): Mezzio\Middleware\LazyLoadingMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #10 /etc/shlink/vendor/laminas/laminas-stratigility/src/MiddlewarePipe.php(76): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
4g4lk #11 /etc/shlink/vendor/laminas/laminas-stratigility/src/Next.php(51): Laminas\Stratigility\MiddlewarePipe->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #12 /etc/shlink/vendor/laminas/laminas-stratigility/src/Middleware/PathMiddlewareDecorator.php(125): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
4g4lk #13 /etc/shlink/vendor/laminas/laminas-stratigility/src/Next.php(44): Psr\Http\Server\RequestHandlerInterface@anonymous->handle(Object(Laminas\Diactoros\ServerRequest))
4g4lk #14 /etc/shlink/module/Rest/src/Middleware/AuthenticationMiddleware.php(52): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
4g4lk #15 /etc/shlink/vendor/mezzio/mezzio/src/Middleware/LazyLoadingMiddleware.php(37): Shlinkio\Shlink\Rest\Middleware\AuthenticationMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #16 /etc/shlink/vendor/laminas/laminas-stratigility/src/Next.php(51): Mezzio\Middleware\LazyLoadingMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #17 /etc/shlink/module/Rest/src/Middleware/BodyParserMiddleware.php(32): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
4g4lk #18 /etc/shlink/vendor/mezzio/mezzio/src/Middleware/LazyLoadingMiddleware.php(37): Shlinkio\Shlink\Rest\Middleware\BodyParserMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #19 /etc/shlink/vendor/laminas/laminas-stratigility/src/Next.php(51): Mezzio\Middleware\LazyLoadingMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #20 /etc/shlink/vendor/mezzio/mezzio-router/src/Middleware/ImplicitOptionsMiddleware.php(72): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
4g4lk #21 /etc/shlink/vendor/mezzio/mezzio/src/Middleware/LazyLoadingMiddleware.php(37): Mezzio\Router\Middleware\ImplicitOptionsMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #22 /etc/shlink/vendor/laminas/laminas-stratigility/src/Next.php(51): Mezzio\Middleware\LazyLoadingMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #23 /etc/shlink/vendor/laminas/laminas-stratigility/src/MiddlewarePipe.php(76): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
4g4lk #24 /etc/shlink/vendor/laminas/laminas-stratigility/src/Middleware/PathMiddlewareDecorator.php(63): Laminas\Stratigility\MiddlewarePipe->process(Object(Laminas\Diactoros\ServerRequest), Object(Psr\Http\Server\RequestHandlerInterface@anonymous))
4g4lk #25 /etc/shlink/vendor/laminas/laminas-stratigility/src/Next.php(51): Laminas\Stratigility\Middleware\PathMiddlewareDecorator->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #26 /etc/shlink/vendor/laminas/laminas-stratigility/src/Next.php(44): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
4g4lk #27 /etc/shlink/vendor/mezzio/mezzio-router/src/Middleware/ImplicitHeadMiddleware.php(76): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
4g4lk #28 /etc/shlink/vendor/mezzio/mezzio/src/Middleware/LazyLoadingMiddleware.php(37): Mezzio\Router\Middleware\ImplicitHeadMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #29 /etc/shlink/vendor/laminas/laminas-stratigility/src/Next.php(51): Mezzio\Middleware\LazyLoadingMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #30 /etc/shlink/vendor/mezzio/mezzio-router/src/Middleware/RouteMiddleware.php(50): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
4g4lk #31 /etc/shlink/vendor/mezzio/mezzio/src/Middleware/LazyLoadingMiddleware.php(37): Mezzio\Router\Middleware\RouteMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #32 /etc/shlink/vendor/laminas/laminas-stratigility/src/Next.php(51): Mezzio\Middleware\LazyLoadingMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #33 /etc/shlink/vendor/laminas/laminas-stratigility/src/MiddlewarePipe.php(76): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
4g4lk #34 /etc/shlink/vendor/laminas/laminas-stratigility/src/Next.php(51): Laminas\Stratigility\MiddlewarePipe->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #35 /etc/shlink/vendor/laminas/laminas-stratigility/src/Next.php(44): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
4g4lk #36 /etc/shlink/vendor/shlinkio/shlink-common/src/Middleware/CloseDbConnectionMiddleware.php(24): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
4g4lk #37 /etc/shlink/vendor/mezzio/mezzio/src/Middleware/LazyLoadingMiddleware.php(37): Shlinkio\Shlink\Common\Middleware\CloseDbConnectionMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #38 /etc/shlink/vendor/laminas/laminas-stratigility/src/Next.php(51): Mezzio\Middleware\LazyLoadingMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #39 /etc/shlink/vendor/laminas/laminas-stratigility/src/MiddlewarePipe.php(76): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
4g4lk #40 /etc/shlink/vendor/laminas/laminas-stratigility/src/Next.php(51): Laminas\Stratigility\MiddlewarePipe->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #41 /etc/shlink/vendor/laminas/laminas-stratigility/src/Middleware/PathMiddlewareDecorator.php(125): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
4g4lk #42 /etc/shlink/vendor/laminas/laminas-stratigility/src/Next.php(44): Psr\Http\Server\RequestHandlerInterface@anonymous->handle(Object(Laminas\Diactoros\ServerRequest))
4g4lk #43 /etc/shlink/vendor/mezzio/mezzio-problem-details/src/ProblemDetailsMiddleware.php(50): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
4g4lk #44 /etc/shlink/vendor/mezzio/mezzio/src/Middleware/LazyLoadingMiddleware.php(37): Mezzio\ProblemDetails\ProblemDetailsMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #45 /etc/shlink/vendor/laminas/laminas-stratigility/src/Next.php(51): Mezzio\Middleware\LazyLoadingMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #46 /etc/shlink/vendor/laminas/laminas-stratigility/src/MiddlewarePipe.php(76): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
4g4lk #47 /etc/shlink/vendor/laminas/laminas-stratigility/src/Middleware/PathMiddlewareDecorator.php(63): Laminas\Stratigility\MiddlewarePipe->process(Object(Laminas\Diactoros\ServerRequest), Object(Psr\Http\Server\RequestHandlerInterface@anonymous))
4g4lk #48 /etc/shlink/vendor/laminas/laminas-stratigility/src/Next.php(51): Laminas\Stratigility\Middleware\PathMiddlewareDecorator->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #49 /etc/shlink/vendor/laminas/laminas-stratigility/src/Next.php(44): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
4g4lk #50 /etc/shlink/module/Rest/src/Middleware/CrossDomainMiddleware.php(25): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
4g4lk #51 /etc/shlink/vendor/mezzio/mezzio/src/Middleware/LazyLoadingMiddleware.php(37): Shlinkio\Shlink\Rest\Middleware\CrossDomainMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #52 /etc/shlink/vendor/laminas/laminas-stratigility/src/Next.php(51): Mezzio\Middleware\LazyLoadingMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #53 /etc/shlink/vendor/laminas/laminas-stratigility/src/Middleware/ErrorHandler.php(131): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
4g4lk #54 /etc/shlink/vendor/mezzio/mezzio/src/Middleware/LazyLoadingMiddleware.php(37): Laminas\Stratigility\Middleware\ErrorHandler->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #55 /etc/shlink/vendor/laminas/laminas-stratigility/src/Next.php(51): Mezzio\Middleware\LazyLoadingMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #56 /etc/shlink/vendor/php-middleware/request-id/src/RequestIdMiddleware.php(32): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
4g4lk #57 /etc/shlink/vendor/mezzio/mezzio/src/Middleware/LazyLoadingMiddleware.php(37): PhpMiddleware\RequestId\RequestIdMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #58 /etc/shlink/vendor/laminas/laminas-stratigility/src/Next.php(51): Mezzio\Middleware\LazyLoadingMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #59 /etc/shlink/vendor/shlinkio/shlink-common/src/Middleware/ContentLengthMiddleware.php(24): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
4g4lk #60 /etc/shlink/vendor/mezzio/mezzio/src/Middleware/LazyLoadingMiddleware.php(37): Shlinkio\Shlink\Common\Middleware\ContentLengthMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #61 /etc/shlink/vendor/laminas/laminas-stratigility/src/Next.php(51): Mezzio\Middleware\LazyLoadingMiddleware->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #62 /etc/shlink/vendor/laminas/laminas-stratigility/src/MiddlewarePipe.php(76): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
4g4lk #63 /etc/shlink/vendor/laminas/laminas-stratigility/src/Next.php(51): Laminas\Stratigility\MiddlewarePipe->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\Next))
4g4lk #64 /etc/shlink/vendor/laminas/laminas-stratigility/src/MiddlewarePipe.php(76): Laminas\Stratigility\Next->handle(Object(Laminas\Diactoros\ServerRequest))
4g4lk #65 /etc/shlink/vendor/laminas/laminas-stratigility/src/MiddlewarePipe.php(65): Laminas\Stratigility\MiddlewarePipe->process(Object(Laminas\Diactoros\ServerRequest), Object(Laminas\Stratigility\EmptyPipelineHandler))
4g4lk #66 /etc/shlink/vendor/mezzio/mezzio-swoole/src/Event/RequestHandlerRequestListener.php(119): Laminas\Stratigility\MiddlewarePipe->handle(Object(Laminas\Diactoros\ServerRequest))
4g4lk #67 /etc/shlink/vendor/mezzio/mezzio-swoole/src/Event/EventDispatcher.php(39): Mezzio\Swoole\Event\RequestHandlerRequestListener->__invoke(Object(Mezzio\Swoole\Event\RequestEvent))
4g4lk #68 /etc/shlink/vendor/mezzio/mezzio-swoole/src/SwooleRequestHandlerRunner.php(135): Mezzio\Swoole\Event\EventDispatcher->dispatch(Object(Mezzio\Swoole\Event\RequestEvent))
4g4lk #69 [internal function]: Mezzio\Swoole\SwooleRequestHandlerRunner->onRequest(Object(Swoole\Http\Request), Object(Swoole\Http\Response))
4g4lk #70 {main}
4g4lk [2022-02-19T21:01:18.850166+00:00] [3d892419-b884-4d60-a829-6d103be3444a] Access.ERROR - - "GET /rest/v2/mercure-info HTTP/1.1" 501 161
```

#### Expected behavior

```shell
4g4lk [2022-02-19T21:01:18.785403+00:00] [e7c0d6ef-d9d0-4c45-9410-c4b71fbbdfc3] Access.INFO - - "GET /rest/v2/health HTTP/1.1" 200 121
4g4lk [2022-02-19T21:01:18.850166+00:00] [3d892419-b884-4d60-a829-6d103be3444a] Access.INFO - - "GET /rest/v2/mercure-info HTTP/1.1" 501 0
```

#### How to reproduce

```yml
version: '3.8'

services:
  shlink:
    image: shlinkio/shlink:latest
    environment:
       DEFAULT_DOMAIN: doma.in
       IS_HTTPS_ENABLED: 'false'
       DB_DRIVER: mysql
       DB_NAME: database
       DB_USER: database
       DB_PASSWORD: password
       DB_HOST: hostname
       DEFAULT_INVALID_SHORT_URL_REDIRECT: http://doma.in
       DEFAULT_REGULAR_404_REDIRECT: http://doma.in
       DEFAULT_BASE_URL_REDIRECT: http://doma.in
       REDIS_SERVERS: tcp://redis:6379
       WEB_WORKER_NUM: 24
       TASK_WORKER_NUM: 24
       GEOLITE_LICENSE_KEY: key
       DEFAULT_SHORT_CODES_LENGTH: 4
       ANONYMIZE_REMOTE_ADDR: 'true'
       REDIRECT_STATUS_CODE: 302
       REDIRECT_CACHE_LIFETIME: 30
       DISABLE_TRACK_PARAM: no-track
    depends_on:
      - redis
    networks:
      - default
      - traefik-public
    deploy:
      mode: replicated
      replicas: 1
      restart_policy:
        condition: any
      labels:
        - traefik.enable=true
        - traefik.http.services.shlink.loadbalancer.server.port=8080
        - traefik.http.routers.shlink-domain.rule=Host(`doma.in`)
        - traefik.http.routers.shlink-domain.entrypoints=http
        - traefik.http.routers.shlink-also-domain.rule=Host(`also-doma.in`)
        - traefik.http.routers.shlink-also-domain.entrypoints=http
        - traefik.docker.network=traefik-public

  redis:
    image: redis:6-alpine
    volumes:
      - shlink-redis-data:/data
    deploy:
      mode: replicated
      replicas: 1
      restart_policy:
        condition: any

volumes:
  shlink-redis-data:
    driver: local

networks:
  traefik-public:
    external: true
```

**Comment by acelaya** [comment]
It's a controlled error, used to tell consumers your instance is not integrated with Mercure. It works like this by design.

https://api-spec.shlink.io/#/Integrations/mercureInfo

The message is logged because the error handler logs as errors all requests resulting in a non 2xx status.

However, it's true that it's a bit noisy. I need to give it a thought to somehow improve this.

**Comment by MarcHagen** [comment]
I'm fine with errors in log dont get me wrong and yes i known it the intended api feature. 

But 40 x 70 lines of error is a bit excessive and somewhat "expensive" on the system (big log file) and/or network usage and processing by log processors.

**Comment by acelaya** [comment]
Yeah, that's what I meant when I was saying it needs improvement.

**Comment by acelaya** [comment]
I have changed the behavior so that the big error message is not logged when this error happens. It will be included in Shlink 3.1.0
